### PR TITLE
[Android] Simplify some operations using Eclipse Collections.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
**Description of the Change**
Use the built-in methods in Eclipse Collections' MutableList and ImmutableList to simplify some operations that would otherwise require complex for loops to be implemented.

**Alternate Designs**
Using Java 8's Streams API and functional interfaces was considered, but their use requires the minimum API level to be 24 (Nougat) or higher ([this will change with Android Studio 4.0](https://developer.android.com/studio/preview/features#j8-desugar)), and Eclipse Collections provides similar simplified operations without using the Streams API.

**Release Notes**
N/A

### CODE REVIEW

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/EventLogActivity.java [hunk 1]
```diff
@@ -44,29 +45,29 @@
...
+    public List<Message> clientLogData = new ArrayList<>();
 
-    public EventLogGuiFragment guiFrag;
     public ListView guiLogList;
     public ArrayAdapter<String> guiLogListAdapter;
-    public ArrayList<String> guiLogData = new ArrayList<>();
+    public List<String> guiLogData = new ArrayList<>();
 
-    private ArrayList<EventLogActivityTabListener<?>> listener = new ArrayList<>();
+    private MutableList<EventLogActivityTabListener<?>> listeners = new FastList<>();
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/EventLogActivity.java:64]: I believe here should be a generic List type, isn't it?

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/PrefsFragment.java [hunk 2]
```diff
@@ -52,19 +61,14 @@
...
-
 public class PrefsFragment extends Fragment {
     private static final String VALUE_LOG = " value: ";
 
     private ListView lv;
     private PrefsListAdapter listAdapter;
 
     // Data for the PrefsListAdapter. This is should be HashMap!
-    private ArrayList<PrefsListItemWrapper> data = new ArrayList<>();
+    private MutableList<PrefsListItemWrapper> data = new FastList<>();
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/PrefsFragment.java:71]: The same generic List?

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/ProjectsFragment.java [hunk 3]
```diff
@@ -60,7 +64,7 @@
 
     private ListView lv;
     private ProjectsListAdapter listAdapter;
-    private ArrayList<ProjectsListData> data = new ArrayList<>();
+    private MutableList<ProjectsListData> data = new FastList<>();
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/ProjectsFragment.java:67]: Generic List?

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/TasksFragment.java [hunk 4]
```diff
@@ -34,23 +26,35 @@
...
+import edu.berkeley.boinc.rpc.Result;
+import edu.berkeley.boinc.rpc.RpcClient;
+import edu.berkeley.boinc.utils.BOINCDefs;
+import edu.berkeley.boinc.utils.Logging;
+
+public class TasksFragment extends Fragment {
     private ListView lv;
     private TasksListAdapter listAdapter;
-    private ArrayList<TaskData> data = new ArrayList<>();
+    private MutableList<TaskData> data = new FastList<>();
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/TasksFragment.java:57]: Generic List?

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.java [hunk 5]
```diff
@@ -35,30 +30,38 @@
...
 
+import edu.berkeley.boinc.BOINCActivity;
+import edu.berkeley.boinc.R;
+import edu.berkeley.boinc.rpc.Project;
+import edu.berkeley.boinc.utils.Logging;
+
 public class NavDrawerListAdapter extends BaseAdapter {
     private Context context;
-    private List<NavDrawerItem> navDrawerItems = new ArrayList<>();
+    private MutableList<NavDrawerItem> navDrawerItems;
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.java:46]: Generic List?

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/client/ClientStatus.java [hunk 6]
```diff
@@ -68,8 +76,8 @@
 
     //RPC wrapper
     private CcStatus status;
-    private List<Result> results;
-    private List<Project> projects;
+    private MutableList<Result> results;
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/client/ClientStatus.java:79]: Why not generic List?

**Comment by AenBleidd** [comment]
Closing and reopening to retrigger CI build

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/EventLogActivity.java (hunk 1 above)
Comment (Isira-Seneviratne) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/EventLogActivity.java:64]: MutableList is an interface that extends the List interface and is part of EC.

Comment (Isira-Seneviratne) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/EventLogActivity.java:64]: https://www.eclipse.org/collections/javadoc/10.2.0/org/eclipse/collections/api/list/MutableList.html

**Pull request merged**